chat-sdk-core/src/main/java/co/chatsdk/core/session/ChatSDK.java

```diff
@@ -92,6 +93,8 @@ public static ChatSDK initialize (Configuration config, BaseNetworkAdapter netwo
             shared().activateModule("FirebaseModule", "activate");
         }
 
+        LocationManager.shared().a = new BaseLocationAdapter();
```

Comment:
Remove this and add a new property inside the ChatSDK class to hold the LocationProvider. i.e. 

```
    protected LocationProvider locationProvider;
```

Provide access via a `locationProvider()` static method.